.github/workflows/test-pr.yml

```diff
@@ -0,0 +1,15 @@
+name: Test PR build
+
+on:
+  workflow_run:
+    workflows: ["Build on PR"]
+    types:
+      - completed
+
+jobs:
+  test:
+    uses: ./.github/workflows/test.yaml
```

Comment:
You probably need something like that around here:
if: ${{ github.event.workflow_run.conclusion == 'success' }}

This workflow would trigger even if the Build failed.